### PR TITLE
ToggleButton: Apply thicker border width to checked outline toggle buttons

### PR DESCRIPTION
**Pull request by khmakoto**
## Current Behavior

Outline `ToggleButtons` look the same when unchecked and when checked, not having any visual distinction between these states.

## New Behavior

Outline `ToggleButtons` have a thicker border when checked (3px) vs when unchecked (1px), having a visual indication between states.

## Related Issue(s)

Fixes part of #21405 and [ADO 14017](https://dev.azure.com/microsoftdesign/fluent-ui/_workitems/edit/14017)

### CODE REVIEW

File: packages/react-components/react-button/src/components/ToggleButton/useToggleButtonStyles.ts [hunk 1]
```diff
@@ -55,6 +55,7 @@ export const useCheckedStyles = makeStyles({
   // Appearance variations
   outline: {
     backgroundColor: tokens.colorTransparentBackgroundSelected,
+    ...shorthands.borderWidth(tokens.strokeWidthThicker),
```
Comment (smhigley) [inline packages/react-components/react-button/src/components/ToggleButton/useToggleButtonStyles.ts:58]: just tested, and unfortunately the focus styles completely override this, so when toggling w/ the keyboard there's no visual change

**Review by smhigley — CHANGES_REQUESTED** [review]
Looks like it'll either need to be implemented with a pseudo element, or need another design look :/

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                      |    Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                  Change |
| :----------------------------------------------------------------------------------------------------- | --------------------------: | ---------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr> | `43.858 kB`<br />`9.973 kB` | `44.189 kB`<br />`10.039 kB` | `331 B` <br />`66 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |   `45.588 kB`<br />`13.32 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |   `32.927 kB`<br />`9.038 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |  `39.854 kB`<br />`10.261 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |    `35.11 kB`<br />`9.784 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          |  `42.369 kB`<br />`11.086 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                                                |   `60.746 kB`<br />`17.41 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                                                          |  `56.031 kB`<br />`16.147 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                                              |    `7.793 kB`<br />`3.327 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                                              |    `9.363 kB`<br />`3.841 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                                            |     `7.765 kB`<br />`3.35 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `181.156 kB`<br />`50.604 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.078 kB`<br />`11.098 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/493c5cf76092b85f9859df6ab378bcbc6e959284'>493c5cf76092b85f9859df6ab378bcbc6e959284</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/Avatar-mount_base.html">1240</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/Avatar-mount.html">1260</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/Button-mount_base.html">740</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/Button-mount.html">770</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/FluentProvider-mount_base.html">2270</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/FluentProvider-mount.html">2336</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/FluentProviderWithTheme-mount_base.html">373</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/FluentProviderWithTheme-mount.html">385</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">307</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">306</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">430</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">433</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/MakeStyles-mount_base.html">2033</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23207/perf-test-react-components/MakeStyles-mount.html">2019</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by smhigley — APPROVED** [review]
It definitely doesn't meet WCAG color alone requirements, but neither do any of the other toggle states, so this one looks fine for now!

**Pull request merged**